### PR TITLE
Added procyon decompiler v0.5.28

### PR DESCRIPTION
**Pull request by asmod3us**
Sending this in to add a formula for [procyon decompiler](https://bitbucket.org/mstrobel/procyon/wiki/Java%20Decompiler).

### CODE REVIEW

File: Library/Formula/procyon-decompiler.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on :java => "1.7+"
+
+  def install
+    libexec.install "procyon-decompiler-#{version}.jar"
+    bin.write_jar_script libexec/"procyon-decompiler-#{version}.jar", "procyon-decompiler", "$@"
+  end
+
+  test do
+    system "#{bin}/procyon-decompiler", "-?"
```
Comment (xu-cheng) [inline Library/Formula/procyon-decompiler.rb]: Can you improve the test to do something more substantial than e.g. `--version` or `--help`? Thanks!

Comment (asmod3us) [inline Library/Formula/procyon-decompiler.rb]: Good point. Actually found a bug while doing this :) I'll update the pull request when it's fixed.

Comment (DomT4) [inline Library/Formula/procyon-decompiler.rb]: Any update here?

**Comment by DomT4** [comment]
Ping?

**Comment by asmod3us** [comment]
Do you have any suggestions for a better test case, preferably something that doesn't require shipping a compiled java class and expected decompiled source?

**Comment by DomT4** [comment]
To be honest, no - I try to avoid Java stuff. Any thoughts @xu-cheng?

**Comment by xu-cheng** [comment]
Write a Java hello world, compile it and then decompile it?

File: Library/Formula/procyon-decompiler.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class ProcyonDecompiler < Formula
+  homepage "https://bitbucket.org/mstrobel/procyon/wiki/Java%20Decompiler"
```
Comment (bfontaine) [inline Library/Formula/procyon-decompiler.rb]: Could you please add a short description above this line using `desc "…"`? Thanks!

**Comment by asmod3us** [comment]
Alright, does this work for you? I've also bumped version to 0.5.29 while I'm at it.

File: Library/Formula/procyon-decompiler.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class ProcyonDecompiler < Formula
+  desc "Procyon Java Decompiler is part of the Procyon suite of Java metaprogramming tools focused on code generation and analysis."
```
Comment (DomT4) [inline Library/Formula/procyon-decompiler.rb]: This is going to be too long. The limit it 80 characters.

Comment (asmod3us) [inline Library/Formula/procyon-decompiler.rb]: I'll shorten it. `brew audit` doesn't seem to capture this :(

Comment (DomT4) [inline Library/Formula/procyon-decompiler.rb]: At the moment we're [only](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/audit.rb#L313) auditing `desc` length during `brew audit --strict`.

Comment (asmod3us) [inline Library/Formula/procyon-decompiler.rb]: Got it, and apparently the name is included in the calculation, so i'll make another attempt.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/db006687ea7b33e87c1131712b072258819febd1. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:.

Just for future reference, ideal commit message for new formulae is styled `procyon-decompiler 0.5.29 (new formula)` or similar, Thanks!

**Pull request closed without merging**